### PR TITLE
Enable more nullable annotations in WebCmdlets

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Enable more nullable comments in WebCmdlets, please review carefully.

## PR Context
Follow up to #19291

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs [hunk 1]
```diff
@@ -112,7 +108,7 @@ public WebCmdletElementCollection Links
                     EnsureHtmlParser();
 
                     List<PSObject> parsedLinks = new();
-                    MatchCollection linkMatch = s_linkRegex.Matches(Content);
+                    MatchCollection linkMatch = s_linkRegex!.Matches(Content!);
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs:111]: "!" says we have bad implementation.  We could use a trick .Net uses - put these regex-s in helper static class so that it is initialized at first use time. I suggest implement this in new PR before.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs [hunk 2]
```diff
@@ -18,12 +20,6 @@ public class BasicHtmlWebResponseObject : WebResponseObject
     {
         #region Constructors
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="BasicHtmlWebResponseObject"/> class.
-        /// </summary>
-        /// <param name="response"></param>
-        public BasicHtmlWebResponseObject(HttpResponseMessage response) : this(response, null) { }
-
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs:26]: Why was this removed? We can not remove public API.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs [hunk 3]
```diff
@@ -96,7 +92,7 @@ public WebCmdletElementCollection Links
                 if (_links == null)
                 {
                     List<PSObject> parsedLinks = new();
-                    MatchCollection linkMatch = HtmlParser.LinkRegex.Matches(Content);
+                    MatchCollection linkMatch = HtmlParser.LinkRegex.Matches(Content!);
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs:95]: Again, `!` usually says that something is wrong.
Please annotate `InitializeContent` method with [MemberNotNull](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.membernotnullattribute)

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs [hunk 4]
```diff
@@ -143,13 +139,13 @@ public WebCmdletElementCollection Images
...
         {
-            string contentType = ContentHelper.GetContentType(BaseResponse);
+            string? contentType = ContentHelper.GetContentType(BaseResponse);
             if (ContentHelper.IsText(contentType))
             {
                 // Fill the Content buffer
                 string characterSet = WebResponseHelper.GetCharacterSet(BaseResponse);
 
-                Content = StreamHelper.DecodeStream(RawContentStream, characterSet, out Encoding encoding);
+                Content = StreamHelper.DecodeStream(RawContentStream!, characterSet, out Encoding encoding);
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs:148]: The same. SetResponse method.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs [hunk 5]
```diff
@@ -118,19 +120,19 @@ private static bool CheckIsText(string contentType)
...
             {
                 // Media types registered with Windows as having a perceived type of text, are text
-                using (RegistryKey contentTypeKey = Registry.ClassesRoot.OpenSubKey(@"MIME\Database\Content Type\" + contentType))
+                using (RegistryKey? contentTypeKey = Registry.ClassesRoot.OpenSubKey(@"MIME\Database\Content Type\" + contentType))
                 {
                     if (contentTypeKey != null)
                     {
-                        string extension = contentTypeKey.GetValue("Extension") as string;
+                        string? extension = contentTypeKey.GetValue("Extension") as string;
                         if (extension != null)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs:128]:
```suggestion
                        if (contentTypeKey.GetValue("Extension") is string extension)
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs [hunk 6]
```diff
@@ -59,19 +61,19 @@ internal static StringBuilder GetRawContentHeader(HttpResponseMessage response)
             return raw;
         }
 
-        internal static bool IsJson(string contentType)
+        internal static bool IsJson(string? contentType)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs:64]: [NotNullWhen](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullwhenattribute)

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs [hunk 7]
```diff
@@ -256,7 +258,7 @@ private static XmlReaderSettings GetSecureXmlReaderSettings()
             return xrs;
         }
 
-        private static bool TryConvertToXml(string xml, out object doc, ref Exception exRef)
+        private static bool TryConvertToXml(string xml, out object? doc, ref Exception? exRef)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs:261]: [NotNullWhen](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullwhenattribute)

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs [hunk 8]
```diff
@@ -278,7 +280,7 @@ private static bool TryConvertToXml(string xml, out object doc, ref Exception ex
             return doc != null;
         }
 
-        private static bool TryConvertToJson(string json, out object obj, ref Exception exRef)
+        private static bool TryConvertToJson(string json, out object? obj, ref Exception? exRef)
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs:283]: [NotNullIfNotNull](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullifnotnullattribute)

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 9]
```diff
@@ -71,13 +72,6 @@ public class WebResponseObject
 
         #region Constructors
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="WebResponseObject"/> class.
-        /// </summary>
-        /// <param name="response"></param>
-        public WebResponseObject(HttpResponseMessage response) : this(response, null)
-        { }
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:79]: Please revert.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObject.cs [hunk 10]
```diff
@@ -46,7 +48,7 @@ public FormObject(string id, string method, string action)
 
         internal void AddField(string key, string value)
         {
-            if (key is not null && !Fields.TryGetValue(key, out string test))
+            if (key is not null && !Fields.TryGetValue(key, out string? test))
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObject.cs:51]:
```suggestion
            if (key is not null && !Fields.TryGetValue(key, out string? _))
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs [hunk 11]
```diff
@@ -433,7 +429,8 @@ internal static bool TryGetEncoding(string characterSet, out Encoding encoding)
             }
             catch (ArgumentException)
             {
-                encoding = null;
+                // Use the default encoding if one wasn't provided
+                encoding = ContentHelper.GetDefaultEncoding();
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs:433]: Please don't change functional changes in the PR.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs (hunk 3 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs]: done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs (hunk 4 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/BasicHtmlWebResponseObject.Common.cs]: done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs (hunk 6 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs]: done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs (hunk 11 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs:433]: I will move it to another PR

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/WebRequestMethod.cs [hunk 12]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+#nullable enable
+
 namespace Microsoft.PowerShell.Commands
 {
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/WebRequestMethod.cs:7]: Please move such trivial files in another PR. This will simplify follow review.
We should annotate base types before.
Aslso I don't see WebRequestSession. It is good candidate for new separate PR.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/WebRequestMethod.cs:7]: I worked on WebRequestSession here: https://github.com/PowerShell/PowerShell/pull/19291

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs (hunk 7 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs]: done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObjectCollection.cs [hunk 13]
```diff
@@ -16,11 +18,11 @@ public class FormObjectCollection : Collection<FormObject>
         /// </summary>
         /// <param name="key"></param>
         /// <returns></returns>
-        public FormObject this[string key]
+        public FormObject? this[string key]
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObjectCollection.cs:21]: The file is not used t all. Please remove in another PR.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObject.cs [hunk 14]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+#nullable enable
+
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObject.cs:5]: It seems FormObject is not used and can be removed in another PR..

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObject.cs:5]: Used in WebRequestPSCmdlet.Common.cs line 1160. Maybe we could remove it anyway

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseObjectFactory.CoreClr.cs [hunk 15]
```diff
@@ -1,6 +1,8 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+#nullable enable
```
[the remark this answers is not in the document]
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseObjectFactory.CoreClr.cs:4]: done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObject.cs (hunk 14 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObject.cs:5]: Yes, it is dead code.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/FormObject.cs:5]: done

**Comment by iSazonov** [comment]
@CarloToso Please resolve merge errors.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 16]
```diff
@@ -20,27 +22,27 @@ public class WebResponseObject
         /// <summary>
         /// Gets or sets the BaseResponse property.
         /// </summary>
-        public HttpResponseMessage BaseResponse { get; set; }
+        public HttpResponseMessage? BaseResponse { get; set; }
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:25]: It seems it is never null.

Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs:25]: It seems it is never null.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs [hunk 17]
```diff
@@ -16,7 +19,7 @@ internal static class ContentHelper
         #region Internal Methods
 
         // ContentType may not exist in response header.  Return null if not.
-        internal static string GetContentType(HttpResponseMessage response) => response.Content.Headers.ContentType?.MediaType;
+        internal static string? GetContentType(HttpResponseMessage? response) => response?.Content.Headers.ContentType?.MediaType;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs:22]: Why is `response` checked for null? It seems it is never null.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs (hunk 8 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs:288]: obj is not null if true.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs [hunk 18]
```diff
@@ -122,26 +122,23 @@ private static bool IsPrintable(char c) => char.IsLetterOrDigit(c)
...
+        [MemberNotNull(nameof(RawContentStream))]
+        private void SetResponse(HttpResponseMessage response, Stream? contentStream)
         {
             ArgumentNullException.ThrowIfNull(response);
 
             BaseResponse = response;
 
-            MemoryStream ms = contentStream as MemoryStream;
+            MemoryStream? ms = contentStream as MemoryStream;
             if (ms is not null)
```
[the remark this answers is not in the document]
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebResponseObject.Common.cs]: done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs (hunk 8 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/InvokeRestMethodCommand.Common.cs]: done

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebProxy.cs [hunk 19]
```diff
@@ -1,14 +1,16 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+#nullable enable
```
[the remark this answers is not in the document]
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebProxy.cs:4]: In another PR

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs (hunk 17 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/ContentHelper.Common.cs]: Probably leftover from a reverted change in a previous commit

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs [hunk 20]
```diff
@@ -419,11 +419,13 @@ internal static string DecodeStream(Stream stream, string characterSet, out Enco
             return content;
         }
 
-        internal static bool TryGetEncoding(string characterSet, out Encoding encoding)
+        internal static bool TryGetEncoding(string? characterSet, out Encoding encoding)
         {
             bool result = false;
             try
             {
+                ArgumentException.ThrowIfNullOrEmpty(characterSet);
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs:427]: Hmm, I think it is wrong. `Encoding.GetEncoding(null)` returns OS default but Encoding.GetEncoding(string.Empty) throw. We should investigate the last case.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs:427]: I think the desired behaviour is that both cases (`characterSet == null`; `characterSet == string.Empty`) should should return `result = false` and `encoding = UTF8` (the default). Do you agree?

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs:427]: The code is in try-catch. So it makes no sense. Please remove.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs:427]: I know it doesn't make sense, I added it to avoid a possible null reference error for characterSet, could point me to a better solution?

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs:427]: Null reference error? In TryGetEncoding? It is impossible since `Encoding.GetEncoding(null)` returns OS default, for string.Empty try-catch will catch and set default too. Also it will set default if .Net doesn't know a value in `characterSet`.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs]: The warning before the last 2 commits was `Possible null reference argument for parameter 'name' in 'Encoding Encoding.GetEncoding(string name)'.`, and the build fails on warnings

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs]: Since [System.Text.Encoding]::GetEncoding($null) return OS default without exception I suggest to open new issue in .Net.
Today we could use a workaround with `!` operator and comment with reference to the new .Net issue.

Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs]: I looked a bit more into it
Test:

```ps1
$source = @"
using System;
using System.Text;


    public class EncodingTest
    {
        public static Encoding GetEncoding1(string name)
        {
            return Encoding.GetEncoding(name);
        }

        public static Encoding GetEncoding2(int name)
        {
            return Encoding.GetEncoding(name);
        }
    }
"@

Add-Type -TypeDefinition $source -Language CSharp -ReferencedAssemblies netstandard, System.Text.Encoding

[EncodingTest]::GetEncoding1($null)
#--> ERROR
[EncodingTest]::GetEncoding2($null)
#--> windows-1252

[EncodingTest]::GetEncoding1("")
#--> ERROR
[EncodingTest]::GetEncoding2("")
#--> windows-1252
[EncodingTest]::GetEncoding2(0)
#--> windows-1252
```

In our case we are using `Encoding.GetEncoding(string name)` so it throws an error with null or empty that gets caught, we don't have a bug everything works as planned. Unfortunalely `Encoding.GetEncoding(string name)` with null gives us a warning, as you said we could use `!` operator.

If `Encoding.GetEncoding(string name)` with name == null returned `windows-1252` that wouldn't be a desired behaviour for us in this case, because we want to default to UTF-8

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseHelper.CoreClr.cs [hunk 21]
```diff
@@ -11,7 +13,7 @@ namespace Microsoft.PowerShell.Commands
 {
     internal static class WebResponseHelper
     {
-        internal static string GetCharacterSet(HttpResponseMessage response) => response.Content.Headers.ContentType?.CharSet;
+        internal static string GetCharacterSet(HttpResponseMessage response) => response.Content.Headers.ContentType?.CharSet ?? string.Empty;
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebResponseHelper.CoreClr.cs:16]: I think we have no need this workaround.

**Comment by CarloToso** [comment]
Unrelated error

**Review by iSazonov — APPROVED** [review]

**Comment by CarloToso** [comment]
@iSazonov can you merge this PR?

**Comment by iSazonov** [comment]
@CarloToso  Please look test fails.

**Comment by CarloToso** [comment]
@iSazonov all tests pass
